packages/customize-widgets/src/components/sidebar-block-editor/use-sidebar-block-editor.js

```diff
@@ -160,107 +158,105 @@ export default function useSidebarBlockEditor( sidebar ) {
 
 					setState( ( lastState ) => {
 						const blocksByWidgetId = keyBy(
-							lastState.blocks,
+							lastState,
 							'attributes.__internalWidgetId'
 						);
 
-						return {
-							...lastState,
-							blocks: widgetIds.map(
-								( widgetId ) => blocksByWidgetId[ widgetId ]
-							),
-						};
+						return widgetIds.map(
+							( widgetId ) => blocksByWidgetId[ widgetId ]
+						);
 					} );
 					break;
 			}
 		};
 
-		sidebar.subscribe( handler );
-		return () => sidebar.unsubscribe( handler );
+		return sidebar.subscribe( handler );
 	}, [ sidebar ] );
 
 	const onChangeBlocks = useCallback(
 		( _nextBlocks ) => {
 			ignoreIncoming.current = true;
 
-			const blocksByWidgetId = keyBy(
-				state.blocks,
-				( block ) => block.attributes.__internalWidgetId
-			);
+			setState( ( lastState ) => {
+				const blocksByWidgetId = keyBy(
+					lastState,
+					( block ) => block.attributes.__internalWidgetId
+				);
 
-			const nextBlocks = _nextBlocks.map( ( nextBlock, index ) => {
-				if (
-					nextBlock.attributes.__internalWidgetId &&
-					nextBlock.attributes.__internalWidgetId in blocksByWidgetId
-				) {
-					const block =
-						blocksByWidgetId[
-							nextBlock.attributes.__internalWidgetId
-						];
-					if ( ! isEqual( block, nextBlock ) ) {
-						const widgetId =
-							nextBlock.attributes.__internalWidgetId;
-						const widgetToUpdate = sidebar.getWidget( widgetId );
-						const widget = blockToWidget(
-							nextBlock,
-							widgetToUpdate
-						);
-						sidebar.updateWidget( widget );
+				const nextBlocks = _nextBlocks.map( ( nextBlock, index ) => {
+					if (
+						nextBlock.attributes.__internalWidgetId &&
+						nextBlock.attributes.__internalWidgetId in
+							blocksByWidgetId
+					) {
+						const block =
+							blocksByWidgetId[
+								nextBlock.attributes.__internalWidgetId
+							];
+						if ( ! isEqual( block, nextBlock ) ) {
+							const widgetId =
+								nextBlock.attributes.__internalWidgetId;
+							const widgetToUpdate = sidebar.getWidget(
+								widgetId
+							);
+							const widget = blockToWidget(
+								nextBlock,
+								widgetToUpdate
+							);
+							sidebar.updateWidget( widget );
+						}
+						return nextBlock;
 					}
-					return nextBlock;
-				}
-
-				const widget = blockToWidget( nextBlock );
-				const widgetId = sidebar.addWidget( widget, index );
-				return {
-					...nextBlock,
-					attributes: {
-						...nextBlock.attributes,
-						__internalWidgetId: widgetId,
-					},
-				};
-			} );
 
-			const seen = nextBlocks.map(
-				( block ) => block.attributes.__internalWidgetId
-			);
+					const widget = blockToWidget( nextBlock );
+					const widgetId = sidebar.addWidget( widget, index );
+					return {
+						...nextBlock,
+						attributes: {
+							...nextBlock.attributes,
```

Comment:
in order not copy inside props we can use[ lodash `merge` ](https://lodash.com/docs/4.17.15#merge)and simplify it:
```js
return merge( nextBlock, {attributes: {____internalWidgetId: widgetId } }); 
```

Reply:
I feel like it's minor as it's only 2 levels deep. I'll definitely consider it if it's more than that though 😅.